sstables/sstables.cc

```diff
@@ -1504,7 +1504,9 @@ future<> seal_summary(summary& s,
     s.header.size = s.entries.size();
     s.header.size_at_full_sampling = sstable::get_size_at_full_sampling(state.partition_count, s.header.min_index_interval);
 
-    assert(first_key); // assume non-empty sstable
+    if (!first_key) {
+        throw malformed_sstable_exception(format("Cannot seal summary because the SSTable has no keys written to it"));
```

Comment:
Please, correct me if I'm wrong, but sealing happens in the context of `table::seal_active_memtable` which doesn't give a damn to what exception pops up unless it's (figuratively speaking) ENOMEM or ENOSPC, so this change would just move the abortion up the stack, not avoid it

Reply:
s/exhausted/expired.